### PR TITLE
fix(connectSearchBox): handle `defaultRefinement`

### LINKED ISSUE #1826: [React] searchBox & defaultRefinement
**Do you want to request a *feature* or report a *bug*?**
Bug

**Bug: What is the current behavior?**
Using `<SearchBox defaultRefinement="hi" />`, the search box remains empty on page load.

**Bug: What is the expected behavior?**
The search box should be initialized with text "hi"

**What project are you opening an issue for?**
react-instantsearch

**What is the version you are using? Always use the latest one before opening a bug issue.**
Latest

### PR DESCRIPTION
**Pull request by vvo**
fixes #1826

### CODE REVIEW

**Comment by mthuret** [comment]
Could you add a story to illustrate on the storybook?

**Comment by vvo** [comment]
> Could you add a story to illustrate on the storybook?

Yep!

**Comment by vvo** [comment]
Done @mthuret

File: packages/react-instantsearch/src/connectors/connectSearchBox.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ function getCurrentRefinement(props, searchState) {
   if (typeof searchState[id] !== 'undefined') {
     return searchState[id];
   }
+  if (typeof props.defaultRefinement === 'string') {
```
Comment (bobylito) [inline packages/react-instantsearch/src/connectors/connectSearchBox.js]: Why not propTypes?

Comment (vvo) [inline packages/react-instantsearch/src/connectors/connectSearchBox.js]: connectSearchBox has no proptypes defined, I will add them. And then only do an `!== undefined` test, works?

Comment (bobylito) [inline packages/react-instantsearch/src/connectors/connectSearchBox.js]: Yep 👍

**Comment by bobylito** [comment]
Will mark mine as waiting for changes, since @vvo proposed some and to not mismerged it :)

**Comment by vvo** [comment]
Woopsie, adding the required changes @bobylito in another PR.

**Pull request merged**